File: 99bottles.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/99_Bottles_of_Beer

def bottles(n)
  return if n == 0
  a = -> (n) { n != 1 ? "s" : "" }
  puts(<<-DOC)
  #{n} bottle#{a[n]} of beer on the wall
  #{n} bottle#{a[n]} of beer
  take one down, pass it around
  #{n-1} bottle#{a[n-1]} of beer on the wall
  
  DOC
  bottles(n-1)
end

bottles(99)

File: 99bottles.pl
#!usr/bin/perl
use strict;
use warnings;

sub n_bottles {
  my $bottle = shift;
  if ($bottle == 1) {
    my $new_bottles = $bottle - 1;
    print "$bottle bottle of beer on the wall\n";
    print "$bottle bottle of beer\n";
    print "take it down, pass it around\n";
    print "$new_bottles bottles of beer\n";
  } else { 
    my $new_bottles = $bottle - 1;
    print "$bottle bottles of beer on the wall\n";
    print "$bottle bottles of beer\n";
    print "take one down, pass it around\n";
    $new_bottles > 1 ? my $bottle_var = "bottles" : my $bottle_var = "bottle"; 
    print "$new_bottles $bottle_var of beer on the wall\n";
    print "......\n";
    return n_bottles($bottle-1);
  }
}

n_bottles(4);
